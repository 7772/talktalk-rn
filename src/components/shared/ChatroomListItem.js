// @flow
import React, { Component } from 'react';
import {
  StyleSheet,
  Image,
  Text,
  View,
  TouchableOpacity,
} from 'react-native';
import moment from 'moment';
import Icon5 from 'react-native-vector-icons/FontAwesome5';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import { Chatroom } from '../../utils/Types';
import { colors } from '../../utils/Styles';

type Styles = {
  container: ViewStyle,
  wrapper: ViewStyle,
  status: ViewStyle,
  imgWrapper: ViewStyle,
  img: ImageStyle,
  viewContent: ViewStyle,
  viewTop: ViewStyle,
  txtDisplayName: TextStyle,
  viewCount: ViewStyle,
  txtCount: TextStyle,
  viewBottom: ViewStyle,
  txtMsg: TextStyle,
  txtDate: TextStyle,
};

const styles: Styles = StyleSheet.create({
  container: {
    width: '100%',
  },
  wrapper: {
    backgroundColor: 'white',
    height: 80,
    borderBottomWidth: 1,
    borderColor: 'rgb(247,248,251)',

    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  status: {
    position: 'absolute',
    width: 12,
    height: 12,
    borderRadius: 6,
    backgroundColor: colors.greenishCyan,
    right: 0,
    bottom: 0,
    borderWidth: 2,
    borderColor: 'white',
  },
  imgWrapper: {
    marginHorizontal: 20,
  },
  img: {
    width: 40,
    height: 40,
  },
  viewContent: {
    flexDirection: 'column',
    flex: 1,
    paddingRight: 20,
  },
  viewTop: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  txtDisplayName: {
    fontSize: 14,
    color: colors.dusk,
  },
  viewCount: {
    width: 16,
    height: 16,
    borderRadius: 8,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgb(80,227,194)',
  },
  txtCount: {
    fontSize: 10,
    color: 'white',
  },
  viewBottom: {
    marginTop: 8,

    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  txtMsg: {
    fontSize: 12,
    color: colors.dusk,
    maxWidth: 150,
  },
  txtDate: {
    fontSize: 12,
    color: colors.blueyGray,
    textAlign: 'right',
  },
});

type Props = {
  style?: ViewStyle;
  item: Chatroom;
  onPress?: () => void;
};

type State = {

};

class Shared extends Component<Props, State> {
  static defaultProps: Props = {
    style: styles.wrapper,
    item: {
      id: '',
      img: null,
      displayName: '',
      msg: '',
      count: '',
      date: null,
      status: false,
    },
  };

  render() {
    return (
      <View style={styles.container}>
        <TouchableOpacity
          activeOpacity={0.5}
          onPress={this.props.onPress}
        >
          <View style={this.props.style}>
            <View style={styles.imgWrapper}>
              {
                this.props.item.img
                  ? <Image style={styles.img} source={this.props.item.img}/>
                  : <Icon5 name="meh" size={40} color={colors.dusk} light/>
              }
              {
                this.props.item.status
                  ? <View style={styles.status}/>
                  : <View/>
              }
            </View>
            <View style={styles.viewContent}>
              <View style={styles.viewTop}>
                <Text style={styles.txtDisplayName}>{this.props.item.displayName}</Text>
                {
                  this.props.item.count
                    ? <View style={styles.viewCount}>
                      <Text style={styles.txtCount}>{this.props.item.count}</Text>
                    </View>
                    : <View/>
                }
              </View>
              <View style={styles.viewBottom}>
                <Text style={styles.txtMsg}>{this.props.item.msg}</Text>
                <Text style={styles.txtDate}>{moment(this.props.item.date).fromNow()}</Text>
              </View>
            </View>
          </View>
        </TouchableOpacity>
      </View>
    );
  }
}

export default Shared;
