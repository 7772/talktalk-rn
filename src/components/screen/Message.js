// @flow
import React, { Component } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  FlatList,
} from 'react-native';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import EmptyListItem from '../shared/EmptyListItem';
import ChatroomListItem from '../shared/ChatroomListItem';
import { Chatroom } from '../../utils/Types';
import { ratio, colors } from '../../utils/Styles';
import { getString } from '../../../STRINGS';

type Styles = {
  container: ViewStyle,
}

const styles: Styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'transparent',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

type Props = {
  navigation: any,
};
type State = {
  rooms: Chatroom[],
};

class Screen extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      rooms: [
        {
          id: 1,
          img: null,
          displayName: 'dooboolab',
          msg: 'When are you finishing??',
          count: 6,
          date: new Date(0),
          status: true,
        },
        {
          id: 2,
          img: null,
          displayName: 'Byun8585',
          msg: 'Hi. This is student from react-native...',
          count: 0,
          date: new Date(0),
          status: false,
        },
      ],
    };
  }

  render() {
    return (
      <View style={styles.container}>
        <FlatList
          style={{
            alignSelf: 'stretch',
          }}
          contentContainerStyle={
            this.state.rooms.length === 0
              ? {
                flex: 1,
                alignSelf: 'stretch',
                alignItems: 'center',
                justifyContent: 'center',
              }
              : null
          }
          keyExtractor={(item, index) => index.toString()}
          data={this.state.rooms}
          renderItem={({ item }) => this.renderItem(item)}
          ListEmptyComponent={<EmptyListItem>{getString('NO_CONTENT')}</EmptyListItem>}
        />
      </View>
    );
  }

  renderItem = (item: Chatroom) => {
    return (
      <ChatroomListItem
        item={item}
        onPress={() => this.onItemClick(item.id)}
      />
    );
  }

  onItemClick = (itemId: string) => {
    this.props.navigation.navigate('Chat', { chatId: itemId });
  }
}

export default Screen;
