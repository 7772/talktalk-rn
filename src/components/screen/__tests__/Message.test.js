import 'react-native';
import * as React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { shallow, render } from 'enzyme';

import Message from '../Message';
import ChatroomListItem from '../../shared/ChatroomListItem';

describe('rendering test', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      navigation: {
        navigate: jest.fn(),
      },
    };
    wrapper = shallow(<Message {...props} />);
  });
  it('should render as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render item', () => {
    const item = [
      {
        id: 1,
        img: null,
        displayName: 'dooboolab',
        msg: 'When are you finishing??',
        count: 6,
        date: new Date(),
        status: true,
      },
      {
        id: 2,
        img: null,
        displayName: 'Byun8585',
        msg: 'Hi. This is student from react-native...',
        count: 0,
        date: new Date(),
        status: false,
      },
    ];
    const rendered = wrapper.instance().renderItem(item);
    expect(wrapper.find(ChatroomListItem).last()).toHaveLength(1);
  });
  it('should navigate on item click', () => {
    wrapper.instance().onItemClick('aaaaaa');
    expect(props.navigation.navigate).toHaveBeenCalledWith('Chat', {"chatId": "aaaaaa"});
  });
});
