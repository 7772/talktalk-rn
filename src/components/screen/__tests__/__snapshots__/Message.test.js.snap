// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering test should render as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <FlatList
    ListEmptyComponent={
      <Shared
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flex": 1,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        No content
      </Shared>
    }
    contentContainerStyle={null}
    data={
      Array [
        Object {
          "count": 6,
          "date": 2019-01-01T03:58:30.311Z,
          "displayName": "dooboolab",
          "id": 1,
          "img": null,
          "msg": "When are you finishing??",
          "status": true,
        },
        Object {
          "count": 0,
          "date": 2019-01-01T03:58:30.311Z,
          "displayName": "Byun8585",
          "id": 2,
          "img": null,
          "msg": "Hi. This is student from react-native...",
          "status": false,
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    style={
      Object {
        "alignSelf": "stretch",
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
